test/cri-containerd/createcontainer_test.go

```diff
@@ -1120,3 +1120,68 @@ func Test_CreateContainer_Mount_NamedPipe_WCOW(t *testing.T) {
 	}
 	runCreateContainerTest(t, wcowHypervisorRuntimeHandler, request)
 }
+
+func Test_CreateContainer_GMSA_WCOW(t *testing.T) {
+	credSpec, err := gmsaSetup(credSpecDefault)
+	if err != nil {
+		t.Skip(err)
```

Comment:
Do we want to skip on error here since we expect to support gmsa for wcow process isolated?

Reply:
Yes, if a cred spec cant be generated the tests cant run/wont function. This wouldn't have anything to do with us supporting it, this is strictly the user doesn't have access to a gmsa account to use on the current machine.

Reply:
I'm more thinking about the scenario where the powershell module is not installed on a test machine or on our local machines, do we want to skip this still then?

Reply:
Yes, the tests cant be run without it. It can't be expected that we have a pre-generated cred spec in a default location for all of us so generating one is necessary.

Reply:
I mean, if the powershell module is not installed, instead of skipping the test, why don't we fail the test with the error returned?

Reply:
I think a test failure should be related to the functionality we're testing not working/failing, not setup that we need to even run the test.

Reply:
The concern I have with that is that we could end up skipping these test when they're run in CloudTest or the type of machine we run the CloudTests on could change, causing us to suddenly start skipping these tests. If we don't closely pay attention to that by checking exactly what tests ran and what tests were skipped for every package generation, we could end up not running these tests for a long time and might miss any breaks introduced.

Reply:
That's whats annoying about this feature. I don't know of a good way (if there is one) to handle this when the tests aren't run locally. The box has to be AD joined and have access to an account there's no way around that.

Reply:
Discussed offline, skipping the test here is fine since we don't currently have a way to setup automatic testing of this in our CloudTest environment anyways.

Reply:
I think we should use the new test feature flag support (currently in PR) here to control whether or not the GMSA tests are run. That would allow us to specify at run time whether we want them to run or not, and then if this setup fails we should fail the test.

Reply:
Thats a good idea :), let me take a look at it so we can get it merged and I can add this

Reply:
@kevpar done